File: frontend/lib/models/project.dart
enum ProjectType {
  flutter,
  web,
  node,
  python,
  other;

  static ProjectType fromString(String value) {
    return ProjectType.values.firstWhere(
      (e) => e.name == value,
      orElse: () => ProjectType.other,
    );
  }
}

class Project {
  final String id;
  final String ownerId;
  final String name;
  final ProjectType type;
  final String rootPath;
  final DateTime createdAt;
  final DateTime updatedAt;

  Project({
    required this.id,
    required this.ownerId,
    required this.name,
    required this.type,
    required this.rootPath,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'],
      ownerId: json['owner_id'],
      name: json['name'],
      type: ProjectType.fromString(json['type']),
      rootPath: json['root_path'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'name': name,
      'type': type.name,
    };
  }
}

File: backend/app/schemas/project.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from enum import Enum


class ProjectType(str, Enum):
    FLUTTER = "flutter"
    WEB = "web"
    NODE = "node"
    PYTHON = "python"
    OTHER = "other"


class ProjectCreate(BaseModel):
    name: str
    type: ProjectType = ProjectType.OTHER


class ProjectResponse(BaseModel):
    id: str
    owner_id: str
    name: str
    type: ProjectType
    root_path: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
